File: zad1.go
package main

import (
	"fmt"
	"math"
)

// Функция для вычисления значения Y в зависимости от X (логика графика)
func calculateY(x float64) float64 {
	if x <= -1 {
		return -x - 2
	} else if x <= 0 {
		return -1 + math.Sqrt(1-x*x)
	} else if x <= 2 {
		return 2 - math.Sqrt(4-x*x)
	} else if x <= 4 {
		return -1
	} else {
		return 0.5*x - 3
	}
}

// Функция ввода-вывода: табулирование функции и построение таблицы
func printTabulatedFunction(xStart, xEnd, dx float64) {
	// Печать шапки таблицы
	fmt.Printf("%10s | %10s\n", "X", "Y")
	fmt.Println("----------------------------")

	// Цикл вычисления и вывода строк таблицы
	for x := xStart; x <= xEnd+dx/2; x += dx {
		y := calculateY(x)
		fmt.Printf("%10.2f | %10.2f\n", x, y)
	}
}

func main() {
	// Входные данные интервала и шага
	xStart := -4.0
	xEnd := 6.0
	dx := 0.5

	// В main остается только вызов управляющей функции
	printTabulatedFunction(xStart, xEnd, dx)
}

File: zad1.cpp
#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

// Функция для вычисления значения Y по графику
double calculate_y(double x) {
    if (x <= -1) {
        return -x - 2;
    } else if (x <= 0) {
        return -1 + sqrt(1 - x * x);
    } else if (x <= 2) {
        return 2 - sqrt(4 - x * x);
    } else if (x <= 4) {
        return -1;
    } else {
        return 0.5 * x - 3;
    }
}

// Функция для вывода таблицы значений
void print_tabulated_function(double x_start, double x_end, double dx) {
    cout << setw(10) << "X" << " | " << setw(10) << "Y" << endl;
    cout << "----------------------------" << endl;
    
    for (double x = x_start; x <= x_end + dx / 2; x += dx) {
        double y = calculate_y(x);
        cout << setw(10) << fixed << setprecision(2) << x 
             << " | " << setw(10) << y << endl;
    }
}

int main() {
    double x_start = -4.0;
    double x_end = 6.0;
    double dx = 0.5;

    // main теперь только запускает процесс
    print_tabulated_function(x_start, x_end, dx);

    return 0;
}